lib/system/io.nim

```diff
@@ -361,6 +365,58 @@ when defined(nimdoc) or (defined(posix) and not defined(nimscript)) or defined(w
       result = setHandleInformation(cast[IoHandle](f), HANDLE_FLAG_INHERIT,
                                     inheritable.WinDWORD) != 0
 
+when defined(nimdoc) or (defined(posix) and not defined(nimscript) and not defined(windows)):
+  proc setNonBlocking*(f: FileHandle, blocking = false) {.raises: [OSError].} =
+    ## Control file handle blocking mode.
+    ##
+    ## Non-blocking IO `read`/`write` calls return immediately with whatever
+    ## result is available, without putting the current thread to sleep. The
+    ## call is expected to be tried again.
+    ##
+    ## Calling `read` on a non-blocking file handle will result in an `IOError`
+    ## of 'Resource temporarily unavailable' whenever there is no data to read.
+    ## The state can be checked beforehand with either
+    ## `endOfFile <#endOfFile,File>`_ or
+    ## `atEnd <streams.html#atEnd,Stream>`_.
+    ##
+    ## This requires the OS file handle, which can be
+    ## retrieved via `getOsFileHandle <#getOsFileHandle,File>`_.
+    ##
+    ## This procedure is available for POSIX platforms. Test for
+    ## availability with `declared() <system.html#declared,untyped>`_.
+    ##
+    ## There are separate APIs on Windows for using console handles,
+    ## pipes and sockets in a non-blocking manner. Some of which aren't
+    ## implemented in stdlib yet.
+    ##
+    ## See `setNonBlocking(File, bool) <#setNonBlocking,File>`_.
+    when not defined(windows):
+      runnableExamples:
```

Comment:
The `when` block should be inside the `runnableExamples` (same for the other proc).